packages/vue-components/src/panels/PanelBase.js

```diff
@@ -175,6 +182,10 @@ export default {
         // Only preload, do not expand the panel.
         return;
       }
+      if (this.showPreview && !this.localExpanded) {
```

Comment:
let's tighten the above branch as well with `localExpanded`, since `wasRetrieverLoaded` was modified

could move to self-explanatory named constants here as well in place of the comments and multiple branches

Reply:
So sorry, I don't quite understand the first part, "tighten the above branch as well with localExpanded, since wasRetrieverLoaded was modified". Can you elaborate more on this? :O

Reply:
this branch `if (this.preloadBool && !this.wasRetrieverLoaded) {`*

you may want to double check with a boolean expression simplifier though

Reply:
I see! Thanks for spotting this :-) I will make the changes!

Reply:
Hmm, after considering your comments and reviewing my code, I realised that all the conditions above are unnecessary and I can just use `(!this.localExpanded)`, since the purpose of the function is to update maxHeight after loading src. 

If the panel is not expanded then, we can simply return the function , regardless of whether it is preload or preview. 

Thanks for the suggestion to improve the code quality anyway 👍